internal/access/credential.go

```diff
@@ -56,7 +56,13 @@ func UpdateCredential(c *gin.Context, user *models.Identity, newPassword string)
 		return err
 	}
 
-	err = checkPasswordRequirements(db, newPassword)
+	return updateCredential(c, user, newPassword, isSelf)
+}
+
+func updateCredential(c *gin.Context, user *models.Identity, newPassword string, isSelf bool) error {
```

Comment:
I had that but it seemed redundant; We don't do anything special to the db objects. the one thing about that pattern is it's ensuring something: you don't get a db object until you've checked authorization, however this function doesn't care about authorization, as it's been checked (or not) by the caller. Since the db is in the ctx it seemed odd to pass both.